### PR TITLE
PlantSeg Integration | AIOD-233

### PR DESCRIPTION
**Pull request by qin-yu**
This PR integrates **PlantSeg v2** into the AI OnDemand (AIoD) framework as a new instance segmentation workflow, with a focus on reliable execution on Nemo and a clear, user-facing entry point.

* Exposes PlantSeg as a user-friendly, end-to-end instance segmentation pipeline in AIoD with three components:
  * neural network–based boundary prediction
  * watershed segmentation
  * GASP agglomeration
* The default exposed output is instance segmentation:
  * Other PlantSeg capabilities (e.g. boundary-only outputs, intermediate masks, interactive proofreading, or fine-tuning) are intentionally not exposed via the AIoD interface at this stage.
* Add PlantSeg conda environment with Nemo TLS workaround
  * Introduces a dedicated PlantSeg environment that installs a patched PlantSeg branch enforcing TLS 1.2.
  * This avoids HTTPS failures caused by TLS 1.3 handshake issues observed on Nemo.
  * Intended as a temporary workaround until the underlying TLS issue is resolved upstream.
* Fix `combine_conda` environment
  * Adds `git` dependency for more reliable execution.

This PR relates to:
* https://github.com/FrancisCrickInstitute/AIoD-Model-Registry/pull/18

### CODE REVIEW

File: modules/models/resources/usr/bin/combine_stacks.py [hunk 1]
```diff
@@ -380,4 +380,10 @@ def mask_iou(masks1: np.ndarray, masks2: np.ndarray):
     aiod_rle.save_encoding(rle=encoded_masks, fpath=save_path)
     # Remove the (symlinked) individual masks now that they are combined
     for mask_path in cli_args.masks:
-        (Path(cli_args.output_dir) / mask_path).unlink()
+        mask_file = Path(mask_path)
```
Comment (sea-shunned) [inline modules/models/resources/usr/bin/combine_stacks.py]: Was there a reason for this change? Removing the mask_path as is at this stage will remove the file form the Nextflow work directory, which breaks the `-resume` functionality. This is why I am only cleaning up the output_dir.

**Review by sea-shunned — CHANGES_REQUESTED** [review]

**Comment by qin-yu** [comment]
The addition of `pydantic` was prompted by a missing-package error I observed when running the Nextflow pipeline with the `local` profile. (shown to @sea-shunned)

After extensive debugging, I found that the issue was caused by user site-packages pollution. In particular, `pydantic` was installed in `~/.local/lib/python3.11/site-packages/`. During Nextflow Conda environment creation, the dependency chain `aiod_utils` → `bioio` → `ome-types` → `pydantic>=2.4.0` led pip to consider `pydantic` already satisfied from the user site, so it skipped installing it into the Conda environment. However, **Nextflow runs processes with `PYTHONNOUSERSITE=true`**, meaning Python cannot see user site-packages and therefore fails to import `pydantic`.

This pollution was likely caused by a previous failed Conda installation, for example due to insufficient space in the home directory during my first few rounds of testing, after which pip continued installing packages into `~/.local`. There are two possible fixes: explicitly add the dependency to the Conda configuration, or remove the user-level installation.

In this case, the pollution was accidental and the user site-packages are not required. The local installation has been removed, and the explicit `pydantic` requirement has therefore been dropped.

For future reference, there is no obvious way to enforce consistent Nextflow behaviour between environment creation and execution. For example, setting `PIP_NO_USER=1` alongside `PYTHONNOUSERSITE=true` only affects execution, while removing `PYTHONNOUSERSITE` makes local testing invalid and prevents reproducing this class of error. Additionally, Conda commands and configuration do not influence pip’s dependency resolution behaviour.

**Comment by qin-yu** [comment]
For future reference:

> This pollution was likely caused by a previous failed Conda installation, for example due to insufficient space in the home directory during my first few rounds of testing, after which pip continued installing packages into `~/.local`. There are two possible fixes: explicitly add the dependency to the Conda configuration, or remove the user-level installation.

After further investigation, this may alternatively be a result of calling `ml` after the Conda environment paths have been prepended. In particular, `ml VirtualGL` is required to launch the AIoD GUI, and loading it at the wrong time may affect the active environment.
